core/base/planarGraphLayout/MergeTreeVisualization.h

```diff
@@ -0,0 +1,930 @@
+    Timer t_scale;
+    if(verbose > 1)
+      printMsg("Scale pairs given persistence", debug::Priority::VERBOSE);
+
+    // printTree(tree);
+    dataType rootPers = tree->getNodePersistence<dataType>(treeRoot);
+
+    std::vector<ftm::idNode> leaves = tree->getLeavesFromTree();
+    auto comp = [&](const ftm::idNode a, const ftm::idNode b) {
+      return tree->getNodePersistence<dataType>(a)
+             < tree->getNodePersistence<dataType>(b);
+    };
+    sort(leaves.begin(), leaves.end(), comp);
+    std::stack<ftm::idNode> stack;
+    for(auto node : leaves)
+      stack.push(node);
+    std::vector<bool> nodeDone(tree->getNumberOfNodes(), false);
+    while(!stack.empty()) {
+      ftm::idNode node = stack.top();
+      stack.pop();
+      nodeDone[node] = true;
+
+      if(node == treeRoot or node == treeRootOrigin or tree->isNodeAlone(node))
+        continue;
+
+      dataType nodePers = tree->getNodePersistence<dataType>(node);
+      ftm::idNode nodeOrigin = tree->getNode(node)->getOrigin();
+
+      // Manage leaf
+      if(tree->isLeaf(node)) {
+        float nodeDiff = (retVec[treeSimplexId[node] * 2]
+                          - retVec[treeSimplexId[nodeOrigin] * 2]);
+        int sign = nodeDiff / std::abs(nodeDiff);
+        auto inc = sign * nodePers / rootPers * (rootYmax - rootYmin) / 2;
+        retVec[treeSimplexId[node] * 2]
+          = retVec[treeSimplexId[nodeOrigin] * 2] + inc;
+
+        // Push nodes in the branch to the stack
+        ftm::idNode nodeParent = tree->getParentSafe(node);
+        ftm::idNode oldNodeParent = -1;
+        while(nodeParent != nodeOrigin) {
+          if(not nodeDone[nodeParent])
+            stack.push(nodeParent);
+          else
+            break;
+          oldNodeParent = nodeParent;
+          nodeParent = tree->getParentSafe(nodeParent);
+          if(oldNodeParent == nodeParent) {
+            std::stringstream ss;
+            ss << "treePlanarLayoutImpl oldNodeParent == nodeParent";
+            printMsg(ss.str(), debug::Priority::VERBOSE);
+            break;
+          }
+        }
+      }
+
+      // Manage saddle
+      if(not tree->isLeaf(node) and not tree->isRoot(node)) {
+        float branchY
+          = retVec[treeSimplexId[tree->getNode(branching[node])->getOrigin()]
+                   * 2];
+        retVec[treeSimplexId[node] * 2] = branchY;
+      }
+    }
+
+    if(verbose > 0) {
+      std::stringstream ss;
+      ss << "SCALE PERS.     = " << t_scale.getElapsedTime();
+      printMsg(ss.str(), debug::Priority::VERBOSE);
+    }
+
+    // ----------------------------------------------------
+    // Branches positionning and avoid edges crossing
+    // ----------------------------------------------------
+    Timer t_avoid;
+    if(verbose > 1)
+      printMsg("Avoid edges crossing", debug::Priority::VERBOSE);
+
+    bool isJT = tree->isJoinTree<dataType>();
+    auto compValue = [&](const ftm::idNode a, const ftm::idNode b) {
```

Comment:
we may add a getScalars to the Merge tree and use its comparison instead here. May be let as a future task

Reply:
Indeed, it could be nice! Ok to leave that as a future task.